lib/controllers/helpers/auth.js

```diff
@@ -10,28 +12,10 @@ var User = mongoose.model("User");
 // Ensures a user is logged in and saves their User object into req.user
 // Should be called in all non-user-registration API requests
 module.exports.authenticate = function authenticate(req, res, next) {
-    var auth = req.headers.authorization;
-
-    // don't authenticate OPTIONS requests for browser compatbility
-    if (req.method === "OPTIONS") return next();
-
-    // must be of the form "Bearer xxx" where xxx is a (variable-length)
-    // access token
-    if (typeof auth === "undefined" || auth.indexOf("Bearer") !== 0) {
-        return next(errors.ACCESS_TOKEN_REQUIRED);
-    }
-
-    // Remove "Bearer " from start of auth
-    var accessToken = auth.slice(7);
-
-    // find user and store in request
-    User.authenticateFromAccessToken(accessToken, function (err, user) {
-        if (err) return next(err);
-
-        req.user = user;
-        // proceed to handle request as normal
-        next();
-    });
+    passport.authenticate("jwt", { session: false }, function(err, user, info) {
```

Comment:
@ElijahGeorge I think this function should actually go in our `passport.js` source file, rather than here. If you look at the structure of the code, the `passport` npm module is already being imported in `passport.js`. The idea is that all of our passport auth functions will be in that source file.

Also, given that we import passport then pass a strategy to it in the other source file, is our passport setup guaranteed to work every time? Just want to make sure.

Reply:
Yes, @jsachs, the passport.js simply serves to configure the application instance of passport and we're carrying out this config by returning it to passportAuth. We then make use of the configured passport module by requiring "passport". I  believe simply requiring passport.js without returning it to passportAuth would have done the same thing. I might do that as it will probably make it more readable. I believe it should work every time as this seems to be the sole location for authentication middleware